### PR TITLE
Mention on a few pages that minlength/maxlength only work on user input

### LINKED ISSUE #4493: Issue with Constraint validation related pages: Not obvious only works on user input
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation
Also applies to: https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation
and maybe other related validation pages

#### What information was incorrect, unhelpful, or incomplete?

There is no mention that the methods `.validity`, `.checkValidity()` and `.reportValidity()` and HTML5 validation in general only work when the user has changed the input directly. For context the `<input type="text">` _does_ include a small one line mention of this under the `minlength` and `maxlength` attributes:

> Constraint validation is only applied when the value is changed by the user

I was tracking down a bug in our app where `.reportValidity()` was not triggering a message even though `minlength` was set. It came down to us using a controlled input which always set the `.value` property of the input on change which meant it didn't count as being changed by the user anymore which in turn meant the validity checks didn't work. It took me a while to find that one sentence on the `<input>` page and the fact that the actual validity pages did _not_ corroborate that fact made me question whether this was the cause until I tested it myself.

#### Specific section or headline?

Maybe under the base "Concepts and usage" section on the API page itself
and maybe under "Constraint validation process" on the second page

#### What did you expect to see?

I would really like to see one of those blue "note" blocks somewhere on the validation pages that says something along the lines of "validation will not have any effect on values set programmatically" and maybe expand that it's only evaluated on user input. I'm not sure the exact wording that's necessary and maybe not even in a callout like the note blocks but there should be _some_ mention of this interaction on those pages in case you miss it on the `<input>` page itself

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/constraint_validation`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/constraint_validation/index.html
* Last commit: https://github.com/mdn/content/commit/d5cf56f90ed1268223b96501149a7ca40b276438
* Document last modified: 2021-02-04T10:39:33.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
Fix https://github.com/mdn/content/issues/4493. This looks like a huge pitfall so I've added it in most places where `minlength`/`maxlength` are mentioned.

### CODE REVIEW

File: files/en-us/learn/forms/form_validation/index.md [hunk 1]
```diff
@@ -244,6 +244,8 @@ A field is invalid if it has a value and that value has fewer characters than th
 Browsers often don't let the user type a longer value than expected into text fields. A better user experience than just using `maxlength` is to also provide character count feedback in an accessible manner and let them edit their content down to size.
 An example of this is the character limit when posting on social media. JavaScript, including [solutions using `maxlength`](https://github.com/mimo84/bootstrap-maxlength), can be used to provide this.
 
+> **Note:** Length constraints are never reported if the value is set programmatically. It's only reported for user-provided input.
```
Comment (chrisdavidmills) [inline files/en-us/learn/forms/form_validation/index.md:247]:
```suggestion
> **Note:** Length constraints are never reported if the value is set programmatically. They are only reported for user-provided input.
```

File: files/en-us/web/html/attributes/minlength/index.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ The length is measured in UTF-16 code units, which ([for most scripts](/en-US/do
 
 The input will fail constraint validation if the length of the text value of the field is less than minlength UTF-16 code units long, with {{domxref('validityState.tooShort')}} returning `true`. Constraint validation is only applied when the value is changed by the user. Once submission fails, some browsers will display an error message indicating the minimum length required and the current length.
 
+`minlength` does not imply [`required`](/en-US/docs/Web/HTML/Attributes/required). An input only violates `minlength` if the user has input something. If an input is not required, it can be empty even if `minlength` is set.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/attributes/minlength/index.md:18]:
```suggestion
`minlength` does not imply [`required`](/en-US/docs/Web/HTML/Attributes/required); an input only violates a `minlength` constraint if the user has input a value. If an input is not `required`, an empty string can be submitted even if `minlength` is set.
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 3]
```diff
@@ -10,7 +10,13 @@ The creation of web forms has always been a complex task. While marking up the f
 
 For a basic introduction to these concepts, with examples, see the [Form validation tutorial](/en-US/docs/Learn/Forms/Form_validation).
 
-> **Note:** HTML Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be sent such as by bad people trying to trick your web application. Therefore, you need to always also validate input constraints on the server side, in a way that is consistent with what is done on the client side.
+> **Note:** HTML Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be sent in many ways:
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:13]:
```suggestion
> **Note:** HTML Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be entered and submitted in many ways:
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 4]
```diff
@@ -10,7 +10,13 @@ The creation of web forms has always been a complex task. While marking up the f
...
+>
+> - By modifying HTML through the inspector
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:15]:
```suggestion
> - By modifying HTML via the browser's developer tools.
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 5]
```diff
@@ -10,7 +10,13 @@ The creation of web forms has always been a complex task. While marking up the f
...
+> - By hand-crafting a request without using the form
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:16]:
```suggestion
> - By hand-crafting an HTTP request without using the form.
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 6]
```diff
@@ -10,7 +10,13 @@ The creation of web forms has always been a complex task. While marking up the f
...
+> - By programmatically writing content to the form (certain constraint validations are _only run_ for user input and not if you set the value of an input in JavaScript)
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:17]:
```suggestion
> - By programmatically writing content into the form (certain constraint validations are _only run_ for user input, and not if you set the value of a form field using JavaScript).
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 7]
```diff
@@ -10,7 +10,13 @@ The creation of web forms has always been a complex task. While marking up the f
...
+>
+> Therefore, you need to always also validate input constraints on the server side, in a way that is consistent with what is done on the client side.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:19]:
```suggestion
> Therefore, you should always validate form data on the server side, consistent with what is done on the client side.
```

File: files/en-us/web/html/constraint_validation/index.md [hunk 8]
```diff
@@ -276,6 +282,7 @@ Calling `checkValidity()` is called _statically_ validating the constraints, whi
 >
 > - If the [`novalidate`](/en-US/docs/Web/HTML/Element/form#novalidate) attribute is set on the {{ HTMLElement("form") }} element, interactive validation of the constraints doesn't happen.
 > - Calling the `submit()` method on the [`HTMLFormElement`](/en-US/docs/Web/API/HTMLFormElement) interface doesn't trigger a constraint validation. In other words, this method sends the form data to the server even if it doesn't satisfy the constraints. Call the `click()` method on a submit button instead.
+> - The `minlength` and `maxlength` constraints are only checked when the input comes from the user. They are not checked even when explicitly calling `checkValidity()` or `reportValidity()` if the value is set programmatically.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/constraint_validation/index.md:285]:
```suggestion
> - The `minlength` and `maxlength` constraints are only checked on user-provided input. They are not checked if a value is set programmatically, even when explicitly calling `checkValidity()` or `reportValidity()`.
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Nice work, @Josh-Cena. I have provided a few grammar suggestions, but nothing major.

File: files/en-us/web/html/constraint_validation/index.md (hunk 3 above)
Comment (Josh-Cena) [reply files/en-us/web/html/constraint_validation/index.md:13]: The current one is better. "Sent" doesn't imply you have to do it through the form, which is consistent with one of the bullet points below.

Comment (chrisdavidmills) [reply files/en-us/web/html/constraint_validation/index.md:13]: Yeah, good point.

**Review by chrisdavidmills — APPROVED** [review]
@Josh-Cena excellent work, thanks! Let's get this merged.

**Pull request merged**